resources/about_storage.md

```diff
@@ -0,0 +1,57 @@
+
+## LMDB
+
+The storage engine of MeiliSearch is [LMDB](http://www.lmdb.tech/doc/). LMDB is a transactional key-value store written in C that was developed for OpenLDAP, and it has ACID properties.
+We choose LMDB after successfully (or not) tried MeiliSearch with [Sled](https://github.com/spacejam/sled) and [RocksDB](https://rocksdb.org/) and decided to move on with LMDB because it is the best combination of performance and stability for Meilisearch.
+
+### Memory mapping
+
+LMDB stores its data in a [memory-mapped file](https://en.wikipedia.org/wiki/Memory-mapped_file). All data fetched from LMDB are returned straight from the memory map, which means there is no memory allocation or memory copy during data fetches.
+
+All documents stored on disk are automatically loaded in memory when MeiliSearch asks for them. This ensures LMDB will always make the best use of the RAM available to retrieve the documents.
+
+For the best performance, it is recommended to provide the same amount of RAM as the size the database takes on disk, so all the data structures can fit in memory.
+
+### Understanding LMDB
+
+The choice of LMDB comes with certain pros and cons. As a result, someone has to understand LMDB database size and memory usage. This is well explained in [a blogpost of LMDB](https://symas.com/understanding-lmdb-database-file-sizes-and-memory-utilization/) and we are trying to summarize it here.
+
+#### Database size
+
+When freeing entries from the database (in our case, removing documents from MeiliSearch), one can observe that no space disk is released. The space the entry took is marked as free for LMDB but not let available for the operating system.
+Unlike other storage engines, LMDB chooses this design for performance issues as there is no need for a compaction phase.
+As a result, you may see that the disk occupied by LMDB and therefore by MeiliSearch keeps growing even when deleting indexes or documents. This is normal behavior, and you can note that the database will not grow again if you delete an index and recreate it after.
+
+#### Memory usage
+
+Since LMDB is memory mapped, it is the operating system that will manage the real memory allocated or not to MeiliSearch.
+Thus, if you run MeiliSearch as a standalone program on a server, LMDB will use the maximum RAM it can uses.
+If you run MeiliSearch along with other programs, the OS will manage memory based on everyone's need making MeiliSearch quite flexible when used in development.
+
+::: tip
+**Virtual memory != Real memory**<br/>
+Virtual memory is the memory asked by a program to the OS. This is not the memory that the program will actually use.
+
+In this case, MeiliSearch will always ask for a memory map of 200Gb. This will always be the virtual memory asked by MeiliSearch to the OS, but as you can see, the amount of RAM used will be less.
```

Comment:
In this case, MeiliSearch will always ask for a memory map of 200Gb. This `refers to` the virtual memory `requested to the OS by MeiliSearch`, but as you can see, the amount of `real memory in` RAM used will be `smaller`.